### ISSUE TITLE
Unknown error 500 on creating event

### ISSUE THREAD

**Issue by iamareebjamal** [fossasia/open-event-server#4654]
**I'm submitting a ...**  (check one with "x")
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-server

**Current behavior:**
Posting this, we get Unknown error 500 but the event is successfully created

```JSON
{
  "data": {
    "attributes": {
      "name": "Test Event",
      "external-event-url": "http://example.com",
      "starts-at": "2017-12-13T23:59:59.123456+00:00",
      "ends-at": "2017-12-14T23:59:59.123456+00:00",
      "timezone": "UTC",
      "latitude": "1.23456789",
      "longitude": "1.23456789",
      "logo-url": "http://example.com/example.png",
      "location-name": "example",
      "searchable-location-name": "example",
      "description": "example",
      "original-image-url": "https://www.w3schools.com/html/pic_mountain.jpg",
      "organizer-name": "example",
      "is-map-shown": "true",
      "organizer-description": "example",
      "is-sessions-speakers-enabled": "true",
      "privacy": "public",
      "state": "published",
      "ticket-url": "http://example.com",
      "code-of-conduct": "example",
      "is-ticketing-enabled": "true",
      "payment-country": "US",
      "payment-currency": "USD",
      "paypal-email": "example@example.com",
      "is-tax-enabled": "true",
      "can-pay-by-paypal": "true",
      "can-pay-by-stripe": "true",
      "can-pay-by-cheque": "true",
      "can-pay-by-bank": "true",
      "can-pay-onsite": "true",
      "cheque-details": "example",
      "bank-details": "example",
      "onsite-details": "example",
      "is-sponsors-enabled": "false",
      "has-organizer-info": "false"
    },
    "type": "event"
  }
}
```

**Expected behavior:**
Either 500 should not be thrown and event should be created properly. Or proper error should be shown and event should not be created 

**Steps to reproduce:**
Create event on open-event-api-dev heroku instance

**Comment by iamareebjamal** [comment]
@srv-twry please check if this issue can be closed

**Comment by srv-twry** [comment]
No, the issue still persists.

``` 
Traceback (most recent call last):
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/__init__.py", line 71, in __call__
    return self.app(environ, start_response)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1540, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/api.py", line 133, in decorated
    return self.check_permissions(view, view_args, view_kwargs, *args, **kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/helpers/permission_manager.py", line 435, in permission_manager
    return permissions[args[0]](view, view_args, view_kwargs, *args, **kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/helpers/permission_manager.py", line 285, in create_event
    return view(*view_args, **view_kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 32, in wrapper
    return func(*args, **kwargs)
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 68, in dispatch_request
    response = method(*args, **kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/decorators.py", line 56, in wrapper
    return func(*args, **kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/resource.py", line 204, in post
    obj = self._data_layer.create_object(data, kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/src/flask-rest-jsonapi/flask_rest_jsonapi/data_layers/alchemy.py", line 51, in create_object
    self.after_create_object(obj, data, view_kwargs)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/events.py", line 117, in after_create_object
    uploaded_images = create_save_image_sizes(data['original_image_url'], 'event', event.id)
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/helpers/files.py", line 159, in create_save_image_sizes
    'original_image_url': create_save_resized_image(image_file, 0, 0, 0, original_upload_path, resize=False),
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/helpers/files.py", line 95, in create_save_resized_image
    temp_file_relative_path = 'static/media/temp/' + generate_hash(str(image_file)) + get_file_name() + '.jpg'
  File "/media/srv_twry/work/Projects/CommunityProjects/Fossasia/open-event-server/app/api/helpers/storage.py", line 251, in generate_hash
    phash = generate_password_hash(key, get_settings()['secret'])
  File "/home/srv_twry/anaconda3/envs/open-event-server/lib/python2.7/site-packages/flask_scrypt.py", line 95, in generate_password_hash
    salt = salt.encode('utf-8')
AttributeError: 'NoneType' object has no attribute 'encode'
```

**Comment by srv-twry** [comment]
@iamareebjamal I found the issue but would need some help here
The problem is that when we are generating the hash for the ```image_url``` file, the salt passed in is ```None```
``` Python
def generate_hash(key):
    """
    Generate hash for key
    """
    phash = generate_password_hash(key, get_settings()['secret'])
    return b64encode(phash)[:10]  # limit len to 10, is sufficient
```

```get_settings()['secret']``` is None

The ```get_settings``` method is as follows
``` Python
def get_settings():
    """
    Use this to get latest system settings
    """
    if 'custom_settings' in current_app.config:
        return current_app.config['custom_settings']
    s = Setting.query.order_by(desc(Setting.id)).first()
    if s is None:
        set_settings(secret='super secret key', app_name='Open Event')
    else:
        current_app.config['custom_settings'] = make_dict(s)
    return current_app.config['custom_settings']
```

The error is due to the line ```current_app.config['custom_settings'] = make_dict(s)``` .

Here is the ```make_dict``` function
``` Python
def make_dict(s):
    arguments = {}
    for name, column in s.__mapper__.columns.items():
        if not (column.primary_key or column.unique):
            arguments[name] = getattr(s, name)
    return arguments
```

So the problem seems to be in the Settings class i.e. the ```settings``` table in the database. It does have a column specified as ```secret``` like 
``` Python
    # App secret
    secret = db.Column(db.String)
``` 
but it seems to be **None** which is causing the whole problem.
How to rectify this ?

**Comment by iamareebjamal** [comment]
That error is not related to this issue. See #4645

**Comment by iamareebjamal** [comment]
See my comments on #4658 and then test if it is solved

**Comment by srv-twry** [comment]
Yes, after using the update command it works. We can close this. 
Please check my PR though, i have linked it to the original issue

**Comment by iamareebjamal** [comment]
Fixed by #4666
